moderator/moderate/auth.py

```diff
@@ -1,84 +1,55 @@
-            for email_entry in self.mozillian_user['alternate_emails']:
-                user_emails.append(email_entry['email'])
-            user_emails.append(self.mozillian_user['email']['value'])
-            users = User.objects.filter(email__in=user_emails)
-            if users:
-                return users[0]
-            return super(ModeratorAuthBackend, self).create_user(claims, **kwargs)
+        # email based filtering
+        users = self.filter_users_by_claims(user_info)
+
+        if len(users) == 1:
+            return self.update_user(users[0], user_info)
+        elif len(users) > 1:
+            # In the rare case that two user accounts have the same email address,
+            # bail. Randomly selecting one seems really wrong.
+            msg = "Multiple users returned"
+            raise SuspiciousOperation(msg)
+        elif self.get_settings("OIDC_CREATE_USER", True):
+            user = self.create_user(user_info)
+            return user
         return None
 
-    def authenticate(self, **kwargs):
-        """Override authenticate method of the OIDC lib."""
-        user = super(ModeratorAuthBackend, self).authenticate(**kwargs)
-        if not user:
-            return None
+    def update_user(self, user, claims):
+        # Update user status (nda, staff based on assertions)
+        groups = claims.get("https://sso.mozilla.com/claim/groups", [])
         profile = user.userprofile
         profile.is_nda_member = False
-
-        try:
-            self.mozillian_user = self.mozillians_client.lookup_user({'email': user.email})
-        except (BadStatusCode, ResourceDoesNotExist):
-            return None
-
-        # The email the user used to log in
-        user_email_domains = [user.email.split('@')[1]]
-        # Get alternate emails
-        for email_resource in self.mozillian_user['alternate_emails']:
-            user_email_domains.append(email_resource['email'].split('@')[1])
-
-        user_email = self.mozillian_user['email'].get('value')
-        if user_email:
-            user_email_domains.append(user_email.split('@')[1])
-        # Remove duplicate domains
-        user_email_domains = list(set(user_email_domains))
-
-        user_groups = [group['name'] for group in self.mozillian_user['groups']['value']]
-
-        # Check if the user is member of the NDA group on each login.
-        # Automatically add users with @mozilla* email in the nda group.
-        if [
-            email_domain
-            for email_domain in user_email_domains
-            if email_domain in settings.TRUSTED_MOZILLA_DOMAINS
-        ] or settings.NDA_GROUP in user_groups:
-            # Find an exact match for the username, eg foo != foo1
+        profile.avatar_url = claims.get("picture")
+        profile.username = claims.get("nickname", "")
+        if (
+            "mozilliansorg_nda" in groups
+            or "mozilliansorg_contigentworkernda" in groups
+        ):
```

Comment:
I think we've lost some functionality here, before if a user was "blah@mozilla.com" they'd have `is_nda_member` set to `True`. Now they won't, they'd have to be in the `mozilliansorg_*nda` group.

Do we envisage allowing non-nda users to use moderator again in the future?

If so I guess in `settings.py` we should do something like:
```
NDA_GROUPS = [
    "team_moco",
    ... etc ...
    "mozilliansorg_contingentworkernda",
]

ALLOWED_LOGIN_GROUPS = NDA_GROUPS + []
```
And here:
```
if set(groups).intersection(settings.NDA_GROUPS):
```

If not, should we just remove the is_nda_member attribute entirely, and remove non-nda behaviour?

Reply:
Good catch!

For now, it makes sense to keep the is_nda_member flag as is. As you correctly mentioned only members of the ALLOWED_LOGIN_GROUPS can login which included the NDA groups. Because of this, the `if` statement can go away altogether and the only thing that's needed is to set `is_nda_member` to True to everyone that logs in.